src/Aspire.Hosting.AWS/CloudFormation/ICloudFormationResource.cs

```diff
@@ -0,0 +1,35 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Amazon.CloudFormation;
+using Amazon.CloudFormation.Model;
+using Aspire.Hosting.ApplicationModel;
+
+namespace Aspire.Hosting.AWS.CloudFormation;
+
+/// <summary>
+/// Resource representing an AWS CloudFormation stack.
+/// </summary>
+public interface ICloudFormationResource : IResource
+{
+    /// <summary>
+    /// Configuration for creating service clients from the AWS .NET SDK.
+    /// </summary>
+    IAWSSDKConfigResource? AWSSDKConfig { get; set; }
```

Comment:
I guess there's a scenario where you specific the region on a per resource basis.

Reply:
It is possible that need to create stacks in multiple regions or accounts. I think the most common is have a default/global `IAWSSDKConfigResource` for an Aspire but I haven't seen how I can automatically add environments variables to all resources of a certain type at the `IDistributedApplicationLifecycleHook` level because then I'm working with the `IResource` not the builder.

Reply:
You can do it, but its a bit more code. https://github.com/dotnet/aspire/blob/85b6275e81af36cd1c974852ce291e319909c4d8/src/Aspire.Hosting.Dapr/DaprDistributedApplicationLifecycleHook.cs#L138